src/index.ts

```diff
@@ -142,46 +145,69 @@ export default class implements ConnectPlugin {
         }
     }
 
-    process(componentConfig: ComponentConfig): Promise<ComponentData> {
-        const links: Link[] = [];
+    get metadata(): Record<string, unknown> {
+        return {
+            storybook: {
+                url: this.targetUrl
+            }
+        };
+    }
 
+    private getStoriesFromStorybook(componentFilePath: string): StorySummary[] {
+        const componentNameFromFilePath = getComponentNameFromFilePath(componentFilePath);
+        return this.stories.filter(story => {
+            const {
+                displayName: storyDisplayName,
+                component,
+                filePath: storyFilePath
+            } = story;
+            return isPathsEqual(componentFilePath, component.filePath) ||
+                isPathsEqual(componentFilePath, storyFilePath) ||
+                component.name === componentNameFromFilePath ||
+                storyDisplayName === componentNameFromFilePath;
+        });
+    }
+
+    private getStoriesFromComponentConfig({
```

Comment:
On second thought, this implementation defeats the purpose of manual configuration. Links should be created even if no story is loaded from targetUrl.

Reply:
Makes sense, I changed a bit with 5d737dd9a1a556059ab76cf452a1229392ae9f9a commit.

Currently, the algorithm tries to find stories from loaded data, if it fails to find then creates URLs with given data.

If the user doesn't set `stories` then it tries to find all stories with the given kind.  If there are no loaded stories with the given kind, then we can't set any storybook URL since we don't know the story name.